### PR TITLE
switch updates

### PR DESCRIPTION
**Pull request by Scrub1492**
Follow-up of #431 .

Switch statements allow for generally easier typing and better performance characteristics. See https://github.com/themoeway/yomitan/pull/431#discussion_r1435925690.

### CODE REVIEW

File: dev/util.js
Comment (toasted-nutbread) [inline dev/util.js:1]: Doesn't really affect this PR, but the custom arg parsing code should really be removed in favor of `util.parseArgs`
https://nodejs.org/api/util.html#utilparseargsconfig

File: ext/js/data/json-schema.js [hunk 1]
```diff
@@ -1343,12 +1343,20 @@ class JsonSchemaProxyHandler {
...
+                if (this._numberPattern.test(property)) {
+                    return Number.parseInt(property, 10);
+                }
+                break;
+            case 'number':
+                if (Math.floor(property) === property && property >= 0) {
+                    return property;
+                }
+                break;
+            default:
```
Comment (toasted-nutbread) [inline ext/js/data/json-schema.js:1357]: default case not needed; omit.

File: ext/js/display/display-generator.js [hunk 2]
```diff
@@ -401,15 +401,22 @@ export class DisplayGenerator {
...
+                    case 'image':
+                        return this._createTermDefinitionEntryImage(entry, dictionary);
+                    case 'structured-content':
+                        return this._createTermDefinitionEntryStructuredContent(entry.content, dictionary);
+                    case 'text':
+                        break;
+                }
+                break;
             }
+            default:
```
Comment (toasted-nutbread) [inline ext/js/display/display-generator.js:418]: default case not needed; omit.

File: ext/js/general/object-property-accessor.js [hunk 3]
```diff
@@ -253,24 +258,31 @@ export class ObjectPropertyAccessor {
...
                     break;
-                case 'close-bracket': // Expecting closing bracket after quoted string
-                    if (v === 0x5d) { // ']'
-                        pathArray.push(value);
-                        value = '';
-                        state = 'next';
-                    } else {
-                        throw new Error(`Unexpected character: ${c}`);
+                case 'close-bracket': { // Expecting closing bracket after quoted string
+                    switch (v) {
```
Comment (toasted-nutbread) [inline ext/js/general/object-property-accessor.js:262]: Don't convert single `if-else` to switch.

File: ext/js/data/json-schema.js (hunk 1 above)
Comment (Scrub1492) [reply ext/js/data/json-schema.js:1357]: It would be preferable to have the default case if we ever turn on the switch-exhaustiveness-check rule though.

Comment (toasted-nutbread) [reply ext/js/data/json-schema.js:1357]: I am not fully convinced it should be on yet, especially given that it doesn't work yet. I'm also not super fond of the idea of needing `default: break;` for things just to make a lint rule happy. We should wait until we can at least get it fully working to evaluate, and maybe only enable it as an opt-in via a meta comment.

**Review by djahandarie — APPROVED** [review]

**Pull request merged**